pkg/operator/ceph/object/cosi/controller.go

```diff
@@ -0,0 +1,291 @@
+		recorder:         mgr.GetEventRecorderFor(controllerName),
+	}
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	controller, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create controller: %v", err)
+	}
+
+	logger.Info("successfully started")
+	// Watch for changes to CephCOSIDriver
+	err = controller.Watch(&source.Kind{Type: &cephv1.CephCOSIDriver{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("failed to watch for CephCOSIDriver object changes: %v", err)
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephCOSIDriver object and makes changes based on the state read
+// and what is in the CephCOSIDriver.Spec
+func (r *ReconcileCephCOSIDriver) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reconcileResponse, cephCOSIDriver, err := r.reconcile(request)
+
+	return reporting.ReportReconcileResult(logger, r.recorder, request, &cephCOSIDriver, reconcileResponse, err)
+}
+
+func (r *ReconcileCephCOSIDriver) reconcile(request reconcile.Request) (reconcile.Result, cephv1.CephCOSIDriver, error) {
+	cephCOSIDriver := &cephv1.CephCOSIDriver{}
+
+	// check whether cosi controller is running
+	cosiControllerDeployment := &appsv1.Deployment{}
+	err := r.client.Get(r.opManagerContext, types.NamespacedName{Name: CosiControllerDeployment, Namespace: CosiControllerNamespace}, cosiControllerDeployment)
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to get cosi controller deployment. %+v", err)
+	}
+
+	// Fetch the CephCOSIDriver instance
+	err = r.client.Get(r.opManagerContext, request.NamespacedName, cephCOSIDriver)
+	if err != nil {
+		if client.IgnoreNotFound(err) == nil {
+			logger.Info("CephCOSIDriver resource not found. Starting with Default values")
+		} else {
+			// Error reading the object - requeue the request.
+			return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to get CephCOSIDriver. %+v", err)
+		}
+	}
+
+	// Set the default CephCOSIDriver name if not already set
+	if cephCOSIDriver.Name == "" {
+		cephCOSIDriver.Name = DefaultCephCOSIDriverName
+	}
+
+	// Set the default CephCOSIDriver namespace if not already set
+	if cephCOSIDriver.Namespace == "" {
+		cephCOSIDriver.Namespace = os.Getenv(k8sutil.PodNamespaceEnvVar)
+	}
+
+	// TODO address non default use cases
+
+	// Check whether object store is running
+	list, err := r.context.RookClientset.CephV1().CephObjectStores(cephCOSIDriver.Namespace).List(r.opManagerContext, metav1.ListOptions{})
+	if err != nil {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("failed to list object stores. %+v", err)
+	}
+
+	// Check for object store part of Multisite
+	if len(list.Items) == 0 {
+		return reconcile.Result{}, *cephCOSIDriver, fmt.Errorf("no object store found, requeuing")
```

Comment:
Let's just log it and not requeue. If an object store is created, we should get another reconcile.
```suggestion
		logger.Infof("no object stores found, skipping cosi driver config")
		return reconcile.Result{}, *cephCOSIDriver, nil
```